src/Symfony/Component/Cache/Adapter/OpCacheAdapter.php

```diff
@@ -0,0 +1,287 @@
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * Adapter building static PHP files that will be cached by OPCache.
+ * This Adapter is in read-only if you use AdapterInterface methods.
+ * You can use the method "store" to build the cache file.
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class OpCacheAdapter implements AdapterInterface, LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    private $file;
+    private $storeSerialized;
+    private $values;
+    private $createCacheItem;
+
+    public function __construct($file, $storeSerialized = true)
+    {
+        if (file_exists($file)) {
+            if (!is_file($file)) {
+                throw new InvalidArgumentException(sprintf('Cache path exists and is not a file: %s', $file));
+            }
+
+            if (!is_writable($file)) {
+                throw new InvalidArgumentException(sprintf('Cache file is not writable: %s', $file));
+            }
+        }
+
+        $directory = dirname($file);
+
+        if (!file_exists($directory)) {
+            @mkdir($directory, 0777, true);
+        }
+
+        if (!file_exists($directory)) {
+            throw new InvalidArgumentException(sprintf('Cache directory does not exist and cannot be created: %s', $directory));
+        }
+
+        if (!is_writable($directory)) {
+            throw new InvalidArgumentException(sprintf('Cache directory is not writable: %s', $directory));
+        }
+
+        $this->file = $file;
+        $this->storeSerialized = $storeSerialized;
+
+        $this->createCacheItem = \Closure::bind(
+            function ($key, $value, $isHit) {
+                $item = new CacheItem();
+                $item->key = $key;
+                $item->value = $value;
+                $item->isHit = $isHit;
+
+                return $item;
+            },
+            $this,
+            CacheItem::class
+        );
+    }
+
+    public function store(array $values = array())
+    {
+        // On PHP 5.6, OPcache is able to keep statically declared arrays in shared memory,
+        // but only up to 32767 elements (See https://bugs.php.net/68057).
+        // On PHP 7.0+, all static arrays are kept in shared memory, statically declared or not.
+        $static = count($values) > 32767 ? '' : 'static ';
+
+        if ($this->storeSerialized) {
+            foreach ($values as $id => $value) {
+                $values[$this->validateKey($id)] = serialize($value);
+            }
+        }
+
+        $exported = var_export($values, true);
+
+        $dump = <<<EOF
+// This file has been auto-generated by the Symfony Cache Component.
+ 
+$static\$staticArray = $exported;
+
+\$array =& \$staticArray;
+unset(\$staticArray);
+
+return \$array;
+EOF;
+
+        // Validates that the dumped code is valid (ie. no __set_state())
+        $this->values = eval($dump);
+
+        $dump = "<?php\n\n" . $dump;
+
+        $newFile = tempnam(dirname($this->file), 'sf_opcache_adapter');
```

Comment:
maybe we should use always the same name, and add locking to prevent concurrent cache file generation ("stampede" protection)